src/sys/node/node-sys.ts

```diff
@@ -99,6 +99,19 @@ export function createNodeSys(c: { process?: any } = {}) {
       opts.window.Response = global.Response;
       opts.window.FetchError = (global as any).FetchError;
     },
+    fetch: (input: any, init: any) => {
+      const nodeFetch = require(path.join(__dirname, 'node-fetch.js'));
```

Comment:
Does it make sense to have a conditional on `global.fetch` like `applyPrerenderGlobalPatch` does? ([Link to function above this one](https://github.com/ionic-team/stencil/pull/2964/commits/d52a7d1134d70fe153dfa9840c313cfc7210eccb#diff-585e96e5364c622963a06a9d71c01a77ad8816bc84e97763dcefe7920176dd50R75)) - That way we don't need to redefine it, right?

Reply:
I'm going to reference this here too: https://github.com/ionic-team/stencil/pull/2964#discussion_r678304924

Looking at this again... Initially I thought we could just call the `applyPrerenderGlobalPatch` function prior to returning the fetch method, but it's signature is designed for the dev server - see lines 81 and 85 where the dev server is hard coded in. Reviewing this, yes I think I should guard this so we don't mandate loading that polyfill code for fetch. 

I switched to a ternary on the first line of this function. That worked great in my test. Good eye here!

Reply:
See https://github.com/ionic-team/stencil/pull/2964/commits/5ca6854a933b5162906f2ea97aba464cff02a7e3 for the ternary

Reply:
Okay I feel great about this, but revisiting https://github.com/ionic-team/stencil/pull/2964#discussion_r678304924, I think I may have to do something over there to consider fetch in the browser or in tests more appropriately.